### ISSUE TITLE
Add Long-Term Memory

### ISSUE THREAD

**Issue by al-yakubovich** [Ngonie-x/langchain_csv#4]
Hi! really awesome project! I just have question about potential usecase.  Let's assume my dataframe has a few columns with textual data. I want to ask a question about which LMM model should have domain-specific knowledge to answer this question. This domain-specific knowledge comes in this case from columns with textual information, and the system is able to find the most relevant cells with text and generate an answer based on this information (let's say my question + text from the top 10 most relevant cells from the dataframe). So for such functionality, we need to have those columns, as well as my question, embedded and put into a vector search database. It would be nice if the project would have such functionality.  So it is essentially how AI assistants with knowledge databases work. I was wondering if it would be possible to add such functionality to the project as well.

**Comment by Ngonie-x** [comment]
Thank you for your insightful comment, I'm glad you liked the project. I find the idea of incorporating domain-specific knowledge from textual columns into our project fascinating. It would certainly enhance the functionality and provide valuable insights.

I will definitely look into the feasibility of implementing such functionality in the project. Would you be willing to collaborate on this issue? Thank you again for your suggestion!

**Comment by al-yakubovich** [comment]
Absolutely! I would love to collaborate! Currently I am doing some research about this.
